### PR TITLE
Add comparison for LogConfig objects

### LINKED ISSUE #549: LogConfig data constructor

### LINKED ISSUE #550: Python comparison for log_config

### PR DESCRIPTION
**Pull request by ManInFez**
Add dict constructor for LogConfig

Fix test of LogConfig to test all constructors

Resolves #550 
Resolves #549

### CODE REVIEW

File: python/res/enkf/log_config.py [hunk 1]
```diff
@@ -61,3 +83,12 @@ def log_file(self):
     @property
     def log_level(self):
         return self._log_level()
+
+    def __eq__(self, other):
+        if self.log_file != other.log_file:
+            print("Files differ {} {}".format(self.log_file, other.log_file))
+            return False
+        if self.log_level != other.log_level:
+            print("level differ {} {}".format(self.log_level, other.log_level))
```
Comment (andreabrambilla) [inline python/res/enkf/log_config.py]: do we really need to have `print`s in a comparison?

File: python/tests/res/enkf/test_log_config.py [hunk 2]
```diff
@@ -50,39 +50,66 @@ def assert_log_config_load(
...
+                    else:
+                        config_dict[ConfigKeys.LOG_LEVEL] = MessageLevelEnum.to_enum(level)
+                else:
+                    config_dict[ConfigKeys.LOG_LEVEL] = MessageLevelEnum.LOG_WARNING
+
+            log_config = LogConfig(user_config_file=self.config_file)
+            res_config = ResConfig(self.config_file)
+            log_config_dict = LogConfig(config_dict=config_dict)
+            self.assertTrue(log_config == log_config_dict)
+            self.assertTrue(log_config == res_config.log_config)
```
Comment (andreabrambilla) [inline python/tests/res/enkf/test_log_config.py]: I think here you could use `assertEqual`

**Review by andreabrambilla — APPROVED** [review]
A couple of minor comments, overall looks good :+1:

File: python/tests/res/enkf/test_log_config.py (hunk 2 above)
Comment (ManInFez) [reply python/tests/res/enkf/test_log_config.py]: I did get a pointer differ error, when I tried this, but trying it again, it seems to work as you are saying.

**Pull request merged**